adal/src/main/java/com/microsoft/aad/adal/ADALError.java

```diff
@@ -120,6 +120,11 @@
     SERVER_ERROR("Server returned an error"),
 
     /**
+     * Server returned the 500/503/504 error.
+     */
+    SERVER_NOT_RESPONDING("Server returned the 500/503/504 error"),
```

Comment:
Hi @kpanwar , do you have any suggestion for the naming? Or how about just creating a new exception class for it, like what we did for UsageAuthenticationException.